src/OrleansAzureUtils/Storage/AzureQueueDataManager.cs

```diff
@@ -140,12 +139,12 @@ public AzureQueueDataManager(string queueName, string deploymentId, string stora
             {
                 // that way we don't have first to create the queue to be able later to delete it.
                 CloudQueue queueRef = queue ?? queueOperationsClient.GetQueueReference(QueueName);
-                var exists = Task<bool>.Factory.FromAsync(queueRef.BeginExists, queueRef.EndExists, null);
-                if (await exists)
+                if (queueRef != null)
```

Comment:
what if `queue `is null? then `queueOperationsClient.GetQueueReference `will not be called? then `queueRef `could be null.